### PR TITLE
docs: Update docs tooling and checks

### PR DESCRIPTION
**Pull request by lornajane**
## What/Why/How?

Following the upgrade on our docs platform, this pull request assembles some updates that were waiting for this change!

Provisional list:

 - [x] fix broken Vale checks on PR (the annotations aren't working due to token permissions)
 - [x] refresh Vale rules and centralise configuration
 - [x] remove outdated jobs, including sync
 - [x] use markdownlint to match destination website repo
 - [x] run prettier on docs folder (as requested in #1111 )
 - [x] revisit link checker
 - [x] add support for running the Redocly CLI docs standalone
 - [x] update Contributing file so that all contributors can use the docs with confidence
 - [x] move Contributing file to top level of repo

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by tatomyr** [comment]
I wonder if it also makes sense to move our contribution guide to the root folder in the scope of this task. I'm not sure why it happens to reside in the .github folder in the first place 🙂

**Comment by lornajane** [comment]
Good idea, I think I will do that. I really prefer to easily find it at the root of the project.

**Review by HCloward — APPROVED** [review]
Wow, very nice! Thanks for doing all of this work.

File: .changeset/bright-steaks-walk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@redocly/cli": patch
+---
+
+feat: Support standalone docs development to facilitate more contributors, add docs quality checks to CI
```
Comment (IgorKarpiuk) [inline .changeset/bright-steaks-walk.md:5]: Should we use sentence case without `type` here ?

```suggestion
Support standalone docs development to facilitate more contributors, add docs quality checks to CI.
```

**Pull request merged**